src/arbitrary.rs

```diff
@@ -388,6 +392,63 @@ impl<T: Arbitrary> Arbitrary for VecDeque<T> {
     }
 }
 
+impl Arbitrary for PathBuf {
+    fn arbitrary<G: Gen>(g: &mut G) -> PathBuf {
+        // use some real directories as guesses, so we may end up with
+        // actual working directories in case that is relevant.
+        let here = std::env::current_dir().unwrap_or(PathBuf::from("/test/directory"));
+        let temp = std::env::temp_dir();
+        let home = std::env::home_dir().unwrap_or(PathBuf::from("/home/user"));
+        let mut p = g.choose(&[here,
+                               temp,
+                               home,
+                               PathBuf::from("."),
+                               PathBuf::from(".."),
+                               PathBuf::from("../../.."),
+                               PathBuf::new()]).unwrap().clone();
+        for sub in Vec::<OsString>::arbitrary(g).iter() {
+            p.push(sub);
+        }
```

Comment:
Seems like this could be replaced with `p.extend(Vec::<OsString>::arbitrary(g).iter())`.